pkg/cloudprovider/aws/apis/v1alpha1/provider_defaults.go

```diff
@@ -75,3 +77,24 @@ func (c *Constraints) defaultSecurityGroups(clusterName string) {
 	}
 	c.SecurityGroupSelector = map[string]string{fmt.Sprintf(ClusterDiscoveryTagKeyFormat, clusterName): "*"}
 }
+
+const (
+	DefaultMetadataOptionsHTTPEndpoint            = ec2.LaunchTemplateInstanceMetadataEndpointStateEnabled
+	DefaultMetadataOptionsHTTPPutResponseHopLimit = 2
+	DefaultMetadataOptionsHTTPTokens              = ec2.LaunchTemplateHttpTokensStateOptional
+)
+
+func (c *Constraints) defaultMetadataOptions() {
+	if c.MetadataOptions == nil {
+		c.MetadataOptions = &MetadataOptions{}
+	}
+	if c.MetadataOptions.HTTPEndpoint == "" {
```

Comment:
If it makes sense for this to potentially be nil, let's make it a ptr

Reply:
I stuggled a bit with this actually. I'm not sure what would work best. It seems like since an empty string would never be a valid value either, it wouldn't matter much if we added the indirection of a pointer. As such, either `""` or `nil` could represent that the user hasn't specified a value and we should pick for them.

I can change this to a pointer though.